### DISCUSSION TITLE
[Question] Is there a way to get docker-gen functionallity with podman?

### DISCUSSION THREAD

**Discussion by aanno2** [jwilder/docker-gen#359] [Q&A]
Hello,

I have used [jwilder/nginx-proxy](https://github.com/jwilder/nginx-proxy) and [JrCs/docker-letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion) a lot that both rely on docker-gen. 

However, with Fedora 31 (and RHEL 8) my distribution has switched to podman (and docker is only useable if you switch back to vgroups v1). With podman, there is no unix socket `/var/run/docker.sock`. However, there seems an replacement for  `docker events`: `podman events` (see [Podman Usage Transfer](https://github.com/containers/libpod/blob/master/transfer.md)).

So I wonder if there is a way to archive the functionallity of docker-gen on podman? 

Kind regards,

aanno2

**Comment by vanaf** [comment]
There is some API https://podman.io/blogs/2020/01/17/podman-new-api.html so maybe docker-gen can be adopted to use it.

**Comment by shyim** [comment]
Tested on podman master branch. It works using podman system service. Should be fixed in next release so of podman

**Comment by karankevich** [comment]
docker-gen generate incorect nginx configuration in section upstream. This section is empty after execute of docker-gen in podman environment. 
I will try to suppose it behavior linked with different result between docker network inspect and podman network inspect. Do somebody know how to get over this issue?

**Comment by jennso** [comment]
> Tested on podman master branch. It works using podman system service. Should be fixed in next release so of podman

Hi I'm trying to implement this on my fedora 33, running podman 2.2.1, could you share me you rconfiguration, I'm not able to get it to work.

**Comment by athisun** [comment]
A similar discussion is happening over on the Traefik repo: https://github.com/traefik/traefik/issues/5730

@jennso SELinux may be preventing your container from connecting to the Podman socket if you're using a rootless container. See here: https://bugzilla.redhat.com/show_bug.cgi?id=1495053#c2

**Comment by heyakyra** [comment]
https://github.com/nginx-proxy/nginx-proxy/discussions/1845#discussioncomment-1902116
